symphony/bdk/core/symphony_bdk.py

```diff
@@ -38,24 +28,15 @@ def __init__(self, config):
 
         self._authenticator_factory = AuthenticatorFactory(config, api_client_factory=self._api_client_factory)
         self._bot_session = AuthSession(self._authenticator_factory.get_bot_authenticator())
-        self._message_service = MessageService(
-            MultiAttachmentsMessagesApi(self._agent_client),
-            MessageApi(self._pod_client),
-            MessageSuppressionApi(self._pod_client),
-            StreamsApi(self._pod_client),
-            PodApi(self._pod_client),
-            AttachmentsApi(self._agent_client),
-            DefaultApi(self._pod_client),
-            self._bot_session
-        )
-        self._datafeed_loop = self._get_datafeed_loop()
-        self._user_service = UserService(
-            UserApi(self._pod_client),
-            UsersApi(self._pod_client),
-            AuditTrailApi(self._agent_client),
-            SystemApi(self._pod_client),
-            self._bot_session
-        )
+
+        self._service_factory = ServiceFactory(self._api_client_factory, self._bot_session, self._config)
+        self._user_service = self._service_factory.get_user_service()
+        self._message_service = self._service_factory.get_message_service()
+        self._connection_service = self._service_factory.get_connection_service()
+        if self._config.is_bot_configured():
+            self._datafeed_loop = self._service_factory.get_datafeed_loop()
+        else:
+            raise BotNotConfiguredException()
```

Comment:
We may have OBO-only bots. So not sure it is relevant.

Reply:
I took this logic from your merged PR for the datafeed loop 😅 https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/blob/2.0/symphony/bdk/core/symphony_bdk.py#L97-L100